### ISSUE TITLE
HP zbook 15u G4 trackpad

### ISSUE THREAD

**Issue by DrdurteeDee** [XXCoreRangerX/Hackintosh-HP-EliteBook-840-G4#10]
I know my device isn't officially supported, but for some reason my trackpad doesn't allow any gestures. Is this normal for the 840's as well or is this a problem of my own making?  

Other than that this EFI works, I only lack the cellular modem and Dgpu.

**Comment by XXCoreRangerX** [comment]
Hello!

I'm reading online that your laptop also has a Synaptics trackpad. Therefore it should work with the same kexts that I'm using.

On my unit the trackpad works with no problem, same for the touchscreen.

---

The kexts I'm using for the trackpad are `VoodooSMBus`, `VoodooRMI` and `VoodooRMI`'s plugin - `RMISMBus`.

1. First, make sure that you have these kexts in `EFI/OC/Kexts`.

2. Then, put them in the following order in `config.plist`:
```xml
     <dict>
       <key>Arch</key>
       <string>Any</string>
       <key>BundlePath</key>
       <string>VoodooInput.kext</string>
       <key>Comment</key>
       <string></string>
       <key>Enabled</key>
       <true/>
       <key>ExecutablePath</key>
       <string>Contents/MacOS/VoodooInput</string>
       <key>MaxKernel</key>
       <string></string>
       <key>MinKernel</key>
       <string></string>
       <key>PlistPath</key>
       <string>Contents/Info.plist</string>
     </dict>
     <dict>
       <key>Arch</key>
       <string>Any</string>
       <key>BundlePath</key>
       <string>VoodooSMBus.kext</string>
       <key>Comment</key>
       <string></string>
       <key>Enabled</key>
       <true/>
       <key>ExecutablePath</key>
       <string>Contents/MacOS/VoodooSMBus</string>
       <key>MaxKernel</key>
       <string></string>
       <key>MinKernel</key>
       <string></string>
       <key>PlistPath</key>
       <string>Contents/Info.plist</string>
     </dict>
     <dict>
       <key>Arch</key>
       <string>Any</string>
       <key>BundlePath</key>
       <string>VoodooRMI.kext</string>
       <key>Comment</key>
       <string></string>
       <key>Enabled</key>
       <true/>
       <key>ExecutablePath</key>
       <string>Contents/MacOS/VoodooRMI</string>
       <key>MaxKernel</key>
       <string></string>
       <key>MinKernel</key>
       <string></string>
       <key>PlistPath</key>
       <string>Contents/Info.plist</string>
     </dict>
     <dict>
       <key>Arch</key>
       <string>Any</string>
       <key>BundlePath</key>
       <string>VoodooRMI.kext/Contents/PlugIns/RMISMBus.kext</string>
       <key>Comment</key>
       <string></string>
       <key>Enabled</key>
       <true/>
       <key>ExecutablePath</key>
       <string>Contents/MacOS/RMISMBus</string>
       <key>MaxKernel</key>
       <string></string>
       <key>MinKernel</key>
       <string></string>
       <key>PlistPath</key>
       <string>Contents/Info.plist</string>
     </dict>
```

**TL;DR:** Kext order: `VoodooInput`, `VoodooSMBus`, `VoodooRMI`, `RMISMBus`.

3. Ensure that any other trackpad related kexts, like VoodooPS2Trackpad, are disabled.
